### ISSUE TITLE
Make it easier to export all form data

### ISSUE THREAD

**Issue by christianbltr** [pagemachine/typo3-formlog#17]
As an editor I want to be able to export all form fields without having to configure them via typoscript because I do not have the rights in my environment to add typoscript configuration.

*Background*
Currently only the basic fields of submitted forms will be exported. More fields can be configured to be exported as described in https://docs.typo3.org/typo3cms/extensions/formlog/ ("Form log export").

Since editors may add their own forms using the visual form editor and they can have access to the formlog module, it would be great if they could export their form data without having to add typoscript configuration.

It would be great if there would be an option to automatically create a column for each property of the "data" object in  the CSV and XLSX export file.

**Comment by mbrodala** [comment]
Generally a good idea but given the fact that all columns of all forms need to be exploded this can get quite ugly. We'd need to traverse all entries once to find out what additional columns need to be exported and then try to get these columns from all form entries, falling back to empty values if absent.

I'll need to give this a bit more thought but welcome any other idea.

**Comment by christianbltr** [comment]
Maybe it would be an suitable approach first let the editor select the page from which he wants to do the export. Only the form log records created by forms sitting on that page then would be exported (mostly there will only be one form on that page). That's the way powermail does it as far as I know. 

Then the number of fields which have to be collected will be significantly reduced.

I think that would be a practical approach because I think it's a seldom use case to export all form log entries from all forms available in the installation. Editors will probably mostly want to export only the data collected by a certain form.

**Comment by yorickreum** [comment]
+1 I'd love this feature too. Or a possibility to simply export all form data, as it is possible with powermail. One can easily clean up with excel afterwards…

**Comment by DenisMir** [comment]
Yeah this would be awesome. We always build the same thing as a custom extension since only viewing the entries is not suitable for most customers.

**Comment by mbrodala** [comment]
@DenisMir Would you like to share what you have built so far?

**Comment by DenisMir** [comment]
@mbrodala Unfortunately the way we build it doesn't fit into the way formlog works. I'm looking into making it be a better fit for formlog.
